bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -464,75 +454,105 @@ private void createSearchTools() {
 		@SuppressWarnings("unused")
 		ToolItem separator = searchTools.createToolItem(SWT.SEPARATOR);
 
+		Runnable searchUpOperation = () -> performSearch(false);
 		searchUpButton = new AccessibleToolItemBuilder(searchTools).withStyleBits(SWT.PUSH)
 				.withImage(FindReplaceOverlayImages.get(FindReplaceOverlayImages.KEY_FIND_PREV))
-				.withToolTipText(FindReplaceMessages.FindReplaceOverlay_upSearchButton_toolTip)
-				.withSelectionListener(SelectionListener.widgetSelectedAdapter(e -> performSearch(false))).build();
+				.withToolTipText(MessageFormat.format(FindReplaceMessages.FindReplaceOverlay_upSearchButton_toolTip,
```

Comment:
Nit: How about extracting the formatting of tooltips into a "addShortcutToToolTipText" method?

Reply:
Maybe even add a "withShortcutHint" method to the ToolItemBuilder, which takes a shortcut and adds it to the ToolTip?

Reply:
Good point! I have added a helper method in ca7b586c04e8743cf7e9e833fa79aceb5a9a676e.
Integrating the shortcut into the tool item (or its builder) would be even better. I am already working on a follow-up cleanup that better integrates shortcuts with operations, as well as the operations assigned to shortcuts and "usual" item clicks, which will cover that proposal in some way. Currently, there is quite some manual effort to initialize everything consistently. But that should not prevent us from having this improvement merged and is also easier to review in separation.